fuel-relayer/src/pending_blocks.rs

```diff
@@ -0,0 +1,617 @@
+        } else {
+            self.handle_block_commit_append(block_root, height, da_height);
+        }
+    }
+
+    async fn bundle(
+        &mut self,
+        to_height: BlockHeight,
+        db: &mut dyn RelayerDb,
+    ) -> Vec<Arc<SealedFuelBlock>> {
+        // if queue is empty check last_finalized_commited fuel block and send all newest ones that we know about.
+        let mut from_height = self.last_commited_finalized_fuel_height;
+        //get blocks range that start from last pending queue item that is not reverted and goes to current block.
+        for pending in self.pending_block_commits.iter() {
+            if !pending.reverted {
+                from_height = pending.block_height;
+                break;
+            }
+        }
+
+        debug!("Bundle from:{from_height}, to:{to_height}");
+        //.expect("This block should be present as we couldn't create new block");
+        //from_height = from_height + BlockHeight::from(1u64);
+        let mut bundle = Vec::new();
+        for height in from_height.as_usize()..=to_height.as_usize() {
+            if let Some(sealed_block) = db.get_sealed_block(BlockHeight::from(height)).await {
+                bundle.push(sealed_block.clone());
+            } else {
+                panic!("All not commited blocks should have its seal and blocks inside db");
+            }
+        }
+        bundle
+    }
+
+    /// When new block is created by this client, bundle all not commited blocks and send it to contract.
+    pub async fn commit<P>(
+        &mut self,
+        height: BlockHeight,
+        db: &mut dyn RelayerDb,
+        provider: &Arc<P>,
+    ) where
+        P: Middleware + 'static,
+    {
+        self.set_chain_height(height);
+        debug!("Handle new created_block {}", height);
+
+        let mut bundle = self.bundle(height, db).await.into_iter();
+
+        let mut parent = if let Some(first_parent) = bundle.next() {
+            first_parent
+        } else {
+            panic!("First Parent should be present:{}", height);
+        };
+
+        for block in bundle {
+            info!(
+                "Bundle send pair {}:{} of blocks:",
+                parent.header.height, block.header.height
+            );
+            if let Err(error) = self.call_contract(&parent, &block, provider).await {
+                warn!("Commit fuel block failed: {}", error);
+                break;
+            }
+            parent = block;
+        }
+    }
+
+    /// Handle commited block from contract.
+    fn handle_block_commit_append(
+        &mut self,
+        block_root: Bytes32,
+        height: BlockHeight,
+        da_height: DaBlockHeight,
+    ) {
+        if self.pending_block_commits.is_empty() {
+            let lcffh = self.last_commited_finalized_fuel_height;
+            if lcffh + 1u64.into() != height {
+                error!("We have missing logs from LCFFH {lcffh} to new height {height}")
```

Comment:
LCFFH is pretty meaningless to a someone debugging the chain, since we never use this abbreviation anywhere else. 

```suggestion
                error!("Missing block commitments between last finalized commitment {lcffh} to new height {height}")
```

Reply:
Reworded in different commit